diff --git a/builtin/settingtypes.txt b/builtin/settingtypes.txt
index dcc5090e4..5d150df73 100644
--- a/builtin/settingtypes.txt
+++ b/builtin/settingtypes.txt
@@ -1906,7 +1906,7 @@ mgvalleys_np_dungeons (Dungeon noise) noise_params_3d 0.9, 0.5, (500, 500, 500),
 
 #    Enable Lua modding support on client.
 #    This support is experimental and API can change.
-enable_client_modding (Client modding) [client] bool false
+enable_client_modding (Client modding) [client] bool true
 
 #    Where to enable server-sent client-side modding (SSCSM).
 #    Warning: Experimental.
diff --git a/src/defaultsettings.cpp b/src/defaultsettings.cpp
index e04e72aa7..940fabd6a 100644
--- a/src/defaultsettings.cpp
+++ b/src/defaultsettings.cpp
@@ -121,7 +121,7 @@ void set_default_settings()
 	settings->setDefault("curl_file_download_timeout", "300000");
 	settings->setDefault("curl_verify_cert", "true");
 	settings->setDefault("enable_remote_media_server", "true");
-	settings->setDefault("enable_client_modding", "false");
+	settings->setDefault("enable_client_modding", "true");
 	settings->setDefault("enable_sscsm", "nowhere");
 	settings->setDefault("max_out_chat_queue_size", "20");
 	settings->setDefault("pause_on_lost_focus", "false");
