diff --git a/src/client/client.cpp b/src/client/client.cpp
index e37cd80d4..eb401cc85 100644
--- a/src/client/client.cpp
+++ b/src/client/client.cpp
@@ -212,11 +212,23 @@ void Client::loadMods()
 		std::string path_user = porting::path_user + DIR_DELIM + "clientmods";
 		const auto modsPath = getClientModsLuaPath();
 		if (modsPath != path_user) {
-			paths["share"] = modsPath;
+			//paths["share"] = modsPath;
+		}
+		//paths["mods"] = path_user;
+
+		const char *env_csm_path = std::getenv("LUANTI_CSM_PATH");
+		Strfnd search_paths(env_csm_path ? env_csm_path : "");
+		while (!search_paths.at_end()) {
+			std::string path = search_paths.next(PATH_DELIM);
+			if (!path.empty())
+				paths["env" + std::to_string(paths.size())] = path;
 		}
-		paths["mods"] = path_user;
 
 		std::string settings_path = path_user + DIR_DELIM + "mods.conf";
+
+		const char *env_csm_conf = std::getenv("LUANTI_CSM_CONFIG");
+		if (env_csm_conf && *env_csm_conf)
+			settings_path = env_csm_conf;
 		modconf.addModsFromConfig(settings_path, paths);
 		modconf.checkConflictsAndDeps();
 	}
