diff --git a/src/server.cpp b/src/server.cpp
index 36133d1b3..8c507b7ee 100644
--- a/src/server.cpp
+++ b/src/server.cpp
@@ -37,6 +37,7 @@
 #include "util/hex.h"
 #include "util/serialize.h"
 #include "util/string.h"
+#include "util/strfnd.h"
 #include "util/thread.h"
 #include "util/tracy_wrapper.h"
 #include "version.h"
@@ -2724,6 +2725,7 @@ void Server::fillMediaCache()
 
 	// ordered in descending priority
 	paths.push_back(getBuiltinLuaPath() + DIR_DELIM "locale");
+	getEnvTextureDirs(paths);
 	fs::GetRecursiveDirs(paths,
 		porting::path_user + DIR_DELIM "textures" DIR_DELIM "server");
 	fs::GetRecursiveDirs(paths,
@@ -2750,6 +2752,15 @@ void Server::fillMediaCache()
 	infostream << "Server: " << m_media.size() << " media files collected" << std::endl;
 }
 
+void Server::getEnvTextureDirs(std::vector<std::string> & paths) const
+{
+	const char *c_texture_path = getenv("LUANTI_TEXTURE_PATH");
+	Strfnd search_paths(c_texture_path ? c_texture_path : "");
+	while (!search_paths.at_end())
+		fs::GetRecursiveDirs(paths, search_paths.next(PATH_DELIM));
+
+}
+
 void Server::sendMediaAnnouncement(session_t peer_id, const std::string &lang_code)
 {
 	auto include = [&] (const std::string &name, const MediaInfo &info) -> bool {
diff --git a/src/server.h b/src/server.h
index 01064e9a8..8a6fee9e3 100644
--- a/src/server.h
+++ b/src/server.h
@@ -344,6 +344,7 @@ class Server : public con::PeerHandler, public MapEventReceiver,
 	const ModSpec* getModSpec(const std::string &modname) const override;
 	const SubgameSpec* getGameSpec() const override { return &m_gamespec; }
 	static std::string getBuiltinLuaPath();
+	void getEnvTextureDirs(std::vector<std::string> &paths) const;
 	std::string getWorldPath() const override { return m_path_world; }
 	std::string getModDataPath() const override { return m_path_mod_data; }
 	ModIPCStore *getModIPCStore() override { return &m_ipcstore; }
