diff --git a/src/server.cpp b/src/server.cpp
index f585e7705..9026cfd4d 100644
--- a/src/server.cpp
+++ b/src/server.cpp
@@ -37,6 +37,7 @@
 #include "util/hex.h"
 #include "util/serialize.h"
 #include "util/string.h"
+#include "util/strfnd.h"
 #include "util/thread.h"
 #include "util/tracy_wrapper.h"
 #include "version.h"
@@ -2667,6 +2668,7 @@ void Server::fillMediaCache()
 
 	// ordered in descending priority
 	paths.push_back(getBuiltinLuaPath() + DIR_DELIM + "locale");
+	getEnvTextureDirs(paths);
 	fs::GetRecursiveDirs(paths, porting::path_user + DIR_DELIM + "textures" + DIR_DELIM + "server");
 	fs::GetRecursiveDirs(paths, m_gamespec.path + DIR_DELIM + "textures");
 	m_modmgr->getModsMediaPaths(paths);
@@ -4001,6 +4003,14 @@ void Server::setAsyncFatalError(const LuaError &e)
 	setAsyncFatalError(std::string("Lua: ") + e.what());
 }
 
+void Server::getEnvTextureDirs(std::vector<std::string> &paths) const
+{
+	const char *c_texture_path = getenv("LUANTI_TEXTURE_PATH");
+	Strfnd search_paths(c_texture_path ? c_texture_path : "");
+	while (!search_paths.at_end())
+		fs::GetRecursiveDirs(paths, search_paths.next(PATH_DELIM));
+}
+
 // Not thread-safe.
 void Server::addShutdownError(const ModError &e)
 {
diff --git a/src/server.h b/src/server.h
index fe21934f9..910d76683 100644
--- a/src/server.h
+++ b/src/server.h
@@ -344,6 +344,7 @@ class Server : public con::PeerHandler, public MapEventReceiver,
 	const ModSpec* getModSpec(const std::string &modname) const override;
 	const SubgameSpec* getGameSpec() const override { return &m_gamespec; }
 	static std::string getBuiltinLuaPath();
+	void getEnvTextureDirs(std::vector<std::string> &paths) const;
 	std::string getWorldPath() const override { return m_path_world; }
 	std::string getModDataPath() const override { return m_path_mod_data; }
 	ModIPCStore *getModIPCStore() override { return &m_ipcstore; }
